apps/opt.c

```diff
@@ -656,22 +659,54 @@ int opt_next(void)
             /* Just a string. */
             break;
         case '/':
-            if (app_isdir(arg) >= 0)
+            if (app_isdir(arg) > 0)
                 break;
             BIO_printf(bio_err, "%s: Not a directory: %s\n", prog, arg);
             return -1;
         case '<':
             /* Input file. */
-            if (strcmp(arg, "-") == 0 || app_access(arg, R_OK) >= 0)
+            if (strcmp(arg, "-") == 0 || app_access(arg, R_OK) == 0)
                 break;
             BIO_printf(bio_err,
                        "%s: Cannot open input file %s, %s\n",
                        prog, arg, strerror(errno));
             return -1;
         case '>':
             /* Output file. */
-            if (strcmp(arg, "-") == 0 || app_access(arg, W_OK) >= 0 || errno == ENOENT)
+            l = strlen(arg);
+            if (arg[l - 1] == '/' || app_isdir(arg) > 0) {
```

Comment:
Hmmm, if the 'intended-output-file' is already existed, it's also necessary to check the writability of this file itself, since a scenario is that the parent directory is writable but the intended file is not writable. So I think there should be 2 times calls for `app_access`.